app/screens/image_preview/downloader.ios.js

```diff
@@ -48,6 +48,16 @@ export default class Downloader extends PureComponent {
         intl: intlShape,
     };
 
+    static getDerivedStateFromProps(props, state) {
+        if (!props.show) {
+            return {
+                didCancel: false,
+                progress: 0,
+            };
+        }
+        return null;
+    }
+
```

Comment:
componentWillReceiveProps needed broken out into two parts - getDerivedStateFromProps for updating our derived state and componentDidUpdate for the side effects.

https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html#side-effects-on-props-change has a good summary about this.